I have reset the branch and excluded the confusing workaround. Now the build for OS X fails:

```
[LD] osx64_build/mixxx-test

Undefined symbols for architecture x86_64:

   "mixxx::AudioSignal::kChannelCountStereo", referenced from:

       mixxx::SoundSourceOpus::tryOpen(mixxx::AudioSourceConfig const&) in soundsourceopus.o

ld: symbol(s) not found for architecture x86_64

clang: error: linker command failed with exit code 1 (use -v to see invocation)

scons: *** [osx64_build/mixxx-test] Error 1

scons: building terminated because of errors.

```

On 17.01.2017 00:48, RJ Skerry-Ryan wrote:
>
> *@rryan* commented on this pull request.
>
> ----------------------------------------------------------------------------
>
> In src/util/audiosignal.cpp <https://github.com/mixxxdj/mixxx/pull/1124>:
>
> > @@ -4,6 +4,31 @@
>   
>   namespace mixxx {
>   
> +// Definitions of static class constants are required by
> +// LLVM CLang, but cause "multiply defined symbols" errors
> +// in Visual Studio 2013. GCC is happy either with or
> +// without those definitions.
> +// TODO(XXX): Replace 'static const' with 'static constexpr'
> +// in the corresponding .h file after upgrading to Visual
> +// Studio 2015 and remove all definitions from this .cpp file.
> +#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
>
> What's confusing is that master builds fine with clang and the static 
> constants defined inline in audiosignal.h and not in audiosignal.o, which 
> contradicts what you are saying about them being mandatory?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/1124>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-altR92BtxZFinJe6-qXzUV968JNEks5rTAHXgaJpZM4Lj3If>.
>

--- a/src/util/audiosignal.cpp
+++ b/src/util/audiosignal.cpp
@@ -4,6 +4,27 @@
 
 namespace mixxx {
 
+// Separate definitions of static class constants are only required
+// for LLVM CLang. Otherwise the linker complains about undefined
+// symbols!?
+#if defined(__clang__)
+const SINT AudioSignal::kChannelCountZero;
+const SINT AudioSignal::kChannelCountDefault;
+const SINT AudioSignal::kChannelCountMono;
+const SINT AudioSignal::kChannelCountMin;
+const SINT AudioSignal::kChannelCountStereo;
+const SINT AudioSignal::kChannelCountMax;
+const SINT AudioSignal::kSamplingRateZero;
+const SINT AudioSignal::kSamplingRateDefault;
+const SINT AudioSignal::kSamplingRateMin;
+const SINT AudioSignal::kSamplingRate32kHz;
+const SINT AudioSignal::kSamplingRateCD;
+const SINT AudioSignal::kSamplingRate48kHz;
+const SINT AudioSignal::kSamplingRate96kHz;
+const SINT AudioSignal::kSamplingRate192kHz;
+const SINT AudioSignal::kSamplingRateMax;
+#endif
+
 bool AudioSignal::verifyReadable() const {
     bool result = true;
     if (!hasValidChannelCount()) {

